File: General/RootViewController.m
//
//  RootViewController.m
//  MICO
//
//  Created by William Xu on 14-5-15.
//  Copyright (c) 2014年 MXCHIP Co;Ltd. All rights reserved.
//

#import "RootViewController.h"
#import "ConThingsViewController.h"
#import "browserViewController.h"
#import <CoreLocation/CoreLocation.h>

@interface RootViewController ()

@property (nonatomic, strong) UIScrollView *scrollView;
@property (nonatomic, strong) CLLocationManager *manager;
@end

@implementation RootViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)awakeFromNib
{
    sleep(1);
    [super awakeFromNib];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    // Segmented control with scrolling
    NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
    NSString *app_Version = [infoDictionary objectForKey:@"CFBundleShortVersionString"];
    self.title = [NSString stringWithFormat:@"My Device Center v%@", app_Version];
    float scrollWidth = self.view.bounds.size.width;
    float scrollHeight =  self.view.bounds.size.height - 44 - 40;
    
    //ceneSegment = [[HMSegmentedControl alloc] initWithSectionTitles:@[@"Home", @"ConThings"]];
    sceneSegment = [[HMSegmentedControl alloc] initWithSectionTitles:@[@"Home"]];
    sceneSegment.autoresizingMask = UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleWidth;
    
    sceneSegment.frame = CGRectMake(0, 0, scrollWidth, 40);
    sceneSegment.segmentEdgeInset = UIEdgeInsetsMake(0, 10, 0, 10);
    sceneSegment.selectionIndicatorHeight = 2.0f;
    sceneSegment.backgroundColor = [UIColor colorWithRed:0.1 green:0.4 blue:0.8 alpha:1];
    
    sceneSegment.titleTextAttributes = @{ NSForegroundColorAttributeName:[UIColor grayColor] };
    sceneSegment.selectedTitleTextAttributes = @{ NSForegroundColorAttributeName:[UIColor whiteColor] };
    
    sceneSegment.selectionIndicatorColor = [UIColor colorWithRed:0.5 green:0.8 blue:1 alpha:1];
    sceneSegment.selectionStyle = HMSegmentedControlSelectionStyleBox;
    sceneSegment.selectionIndicatorLocation = HMSegmentedControlSelectionIndicatorLocationUp;
    //sceneSegment.scrollEnabled = YES;
    [sceneSegment addTarget:self action:@selector(segmentedControlChangedValue:) forControlEvents:UIControlEventValueChanged];
    [self.view addSubview:sceneSegment];
    
    self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectMake(0, 40, scrollWidth, scrollHeight)]; //40 segcontrol
    self.scrollView.backgroundColor = [UIColor colorWithRed:0.7 green:0.7 blue:0.7 alpha:1];
    self.scrollView.pagingEnabled = YES;
    self.scrollView.showsHorizontalScrollIndicator = NO;
    self.scrollView.contentSize = CGSizeMake(scrollWidth, scrollHeight);
    self.scrollView.delegate = self;
    [self.scrollView scrollRectToVisible:CGRectMake(0, 0, scrollWidth, scrollHeight) animated:YES];
    [self.view addSubview:self.scrollView];

    //ConThingsViewController *ConThings = [self.storyboard instantiateViewControllerWithIdentifier:@"ConThings"];
    browserViewController *localDevice = [self.storyboard instantiateViewControllerWithIdentifier:@"Local Device"];
    //ConThingsViewController *conThings = [self.storyboard instantiateViewControllerWithIdentifier:@"Local Device"];
    
    /*Local devices list*/
    localDevice.view.frame = CGRectMake(0, 0, scrollWidth, scrollHeight);
    [localDevice willMoveToParentViewController:self];
    [self.scrollView addSubview:localDevice.view];
    [self addChildViewController:localDevice];
    [localDevice didMoveToParentViewController:self];
    
    
    /*Devices list on www.conthings.com*/
//    conThings.view.frame = CGRectMake(320, 0, 320, 464);
//    
//    [conThings willMoveToParentViewController:self];
//    [self.scrollView addSubview:conThings.view];
//    [self addChildViewController:conThings];
//    [conThings didMoveToParentViewController:self];
    self.manager = [CLLocationManager new];
    [self.manager requestWhenInUseAuthorization];
}

- (void)segmentedControlChangedValue:(HMSegmentedControl *)segmentedControl {
	NSLog(@"Selected index %ld (via UIControlEventValueChanged)", (long)segmentedControl.selectedSegmentIndex);
    [self.scrollView scrollRectToVisible:CGRectMake(segmentedControl.selectedSegmentIndex*320, 0, 320, 500) animated:YES];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction) guideButtonPressed: (UIButton *) button
{
    NSString *textURL = @"http://www.mxchip.com/mico/begin/micoapis/";
    NSURL *cleanURL = [NSURL URLWithString:[NSString stringWithFormat:@"%@", textURL]];
    [[UIApplication sharedApplication] openURL:cleanURL];
}

#pragma mark - UIScrollViewDelegate

- (void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView {
    CGFloat pageWidth = scrollView.frame.size.width;
    NSInteger page = scrollView.contentOffset.x / pageWidth;
    
    [sceneSegment setSelectedSegmentIndex:page animated:YES];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: General/RootViewController.swift
//
//  RootViewController.swift
//  MICO
//
//  Created by William Xu on 2020/5/11.
//  Copyright © 2020 MXCHIP Co;Ltd. All rights reserved.
//

import UIKit
import CoreLocation

let sceneSegmentHeight = CGFloat(40)

class RootViewController: UIViewController, UIScrollViewDelegate {
    
    let manager: CLLocationManager = CLLocationManager()
    var scrollView: UIScrollView!
    var scrollWidth, scrollHeight : CGFloat!
    
    override func awakeFromNib() {
        sleep(1)
        super.awakeFromNib()
    }
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        scrollWidth = UIScreen.main.bounds.size.width
        scrollHeight =  UIScreen.main.bounds.size.height - (self.navigationController?.navigationBar.bounds.size.height ?? CGFloat(44)) - sceneSegmentHeight
        
        
        let version = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "1.0.0"
        self.title = "My Device Center v\(version)"
        
        let sceneSegment: HMSegmentedControl! = HMSegmentedControl.init(sectionTitles: ["Wi-Fi", "BT Mesh"])
        sceneSegment.autoresizingMask = [.flexibleRightMargin, .flexibleWidth]
        sceneSegment.frame = CGRect(x: 0, y: 0, width: scrollWidth, height: sceneSegmentHeight)
        sceneSegment.segmentEdgeInset = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 10)
        sceneSegment.selectionIndicatorHeight = 2.0
        sceneSegment.backgroundColor = UIColor(red: 0.1, green: 0.4, blue: 0.8, alpha: 1)
        sceneSegment.titleTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor.gray]
        sceneSegment.selectedTitleTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor.white]
        sceneSegment.selectionIndicatorColor = UIColor(red: 0.5, green: 0.8, blue: 1, alpha: 1)
        sceneSegment.selectionStyle = .box
        sceneSegment.selectionIndicatorLocation = .up
        sceneSegment.addTarget(self, action: #selector(segmentedControlChangedValue), for: .valueChanged)
        self.view.addSubview(sceneSegment)
        
        scrollView = UIScrollView.init(frame: CGRect(x: 0, y: sceneSegmentHeight, width: scrollWidth, height: scrollHeight))
        scrollView.backgroundColor = UIColor(red: 0.7, green: 0.7, blue: 0.7, alpha: 1)
        scrollView.isPagingEnabled = true
        scrollView.showsHorizontalScrollIndicator = false
        scrollView.contentSize = CGSize(width: scrollWidth * 2, height: scrollHeight)
        scrollView.delegate = self;
        scrollView.scrollRectToVisible(CGRect(x: 0, y: 0, width: scrollWidth, height: scrollHeight), animated: true)
        self.view.addSubview(scrollView)
        
        let localDevice = self.storyboard!.instantiateViewController(withIdentifier: "Wi-Fi") as! browserViewController
        let conThings = self.storyboard!.instantiateViewController(withIdentifier: "BTMesh") as! ConThingsViewController
        
        /*Local devices list*/
        localDevice.view.frame = CGRect(x: 0, y: 0, width: scrollWidth, height: scrollHeight)
        //localDevice.willMove(toParent: self)
        scrollView.addSubview(localDevice.view)
        self.addChild(localDevice)
        //localDevice.didMove(toParent: self)
        
        /*Devices list on www.conthings.com*/
        conThings.view.frame = localDevice.view.frame.offsetBy(dx: scrollWidth, dy: 0)
        //conThings.willMove(toParent: self)
        scrollView.addSubview(conThings.view)
        self.addChild(conThings)
        //conThings.didMove(toParent: self)
        
        manager.requestWhenInUseAuthorization()
        
    }
    
    @objc func segmentedControlChangedValue(_ segmentedControl: HMSegmentedControl) {
        let scrollRect = UIScreen.main.bounds.offsetBy(dx: CGFloat(segmentedControl.selectedSegmentIndex) * scrollWidth, dy: 0)
        self.scrollView.scrollRectToVisible(scrollRect, animated: true)
    }

    @IBAction func guideButtonPressed(_ button: UIButton) {
        NSLog("guideButtonPressed")
        if let guildURL = URL.init(string: "http://www.mxchip.com") {
            UIApplication.shared.open(guildURL)
        }
    }

//    - (void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView {
//        CGFloat pageWidth = scrollView.frame.size.width;
//        NSInteger page = scrollView.contentOffset.x / pageWidth;
//        
//        [sceneSegment setSelectedSegmentIndex:page animated:YES];
//    }
    
}
